pkg/util/lsof/lsof_windows.go

```diff
@@ -5,6 +5,121 @@
 
 package lsof
 
-func openFiles(_ int) (Files, error) {
-	return nil, ErrNotImplemented
+import (
+	"fmt"
+	"os"
+	"unsafe"
+
+	"golang.org/x/sys/windows"
+)
+
+// permissions required to read DLLs loaded by another process
+//
+//nolint:revive
+const (
+	PROCESS_QUERY_INFORMATION = 0x0400
+	PROCESS_VM_READ           = 0x0010
+)
+
+func openFiles(pid int) (Files, error) {
+	handle, err := windows.OpenProcess(PROCESS_QUERY_INFORMATION|PROCESS_VM_READ, false, uint32(pid))
+	if err != nil {
+		return nil, err
+	}
+	defer windows.Close(handle)
+
+	ofl := &openFilesLister{
+		windows.EnumProcessModules,
+		windows.GetModuleFileNameEx,
+		os.Stat,
+	}
+
+	files, err := ofl.getDLLFiles(handle)
+
+	return files, err
+}
+
+type openFilesLister struct {
+	EnumProcessModules  func(windows.Handle, *windows.Handle, uint32, *uint32) error
+	GetModuleFileNameEx func(windows.Handle, windows.Handle, *uint16, uint32) error
+	Stat                func(string) (os.FileInfo, error)
+}
+
+// getDLLFiles returns the list of File for the DLLs opened by the process
+func (ofl *openFilesLister) getDLLFiles(process windows.Handle) (Files, error) {
+	hModules, err := ofl.listOpenDLL(process)
+	if err != nil {
+		return nil, err
+	}
+
+	var files Files
+	for _, hModule := range hModules {
+		file, err := ofl.getDLLFile(process, hModule)
+		if err != nil {
+			continue
```

Comment:
How do you handle file you cannot read?

Reply:
The only case this can return an error is if the `GetModuleFileNameEx` syscall fails, which I assumed meant the module had been removed since
Do you know if it can fail due to permission issues ?

Reply:
The [doc](https://learn.microsoft.com/en-us/windows/win32/api/psapi/nf-psapi-getmodulefilenameexa#return-value) doesn't explain how the call can fail...

Reply:
Ok that's what I was thinking, I'll update it !

Reply:
Done 👍